src/commands/cdn/upload.js

```diff
@@ -1,9 +1,38 @@
-const { Command, flags } = require('@oclif/command');
+const { flags } = require('@oclif/command');
 
-class UploadCommand extends Command {
+const BaseCommand = require('../../utils/base-command');
+const { getFilesAndDir, createTarball } = require('../../utils/file');
+
+/* Class representing the cdn:upload command */
+class UploadCommand extends BaseCommand {
+  /**
+   * Execute the create command
+   * @returns {Promise<void>} Promise representing the command execution result
+   */
   async run() {
-    // TBD
-    this.log('This is not implemented yet');
+    const cmd = this.parse(UploadCommand);
+    const {
+      flags: { expiration },
+      args: { filepath }
+    } = cmd;
+
+    const { files, dir } = await getFilesAndDir(filepath);
+    const { tarPath, deleteTarball } = await createTarball(dir, files);
+
+    try {
+      const result = await this._request.uploadFile({
+        endpoint: '/cdn',
+        filepath: tarPath,
+        query: { expiration }
+      });
+
+      this.log(JSON.stringify(result, null, 2));
+    } catch (err) {
+      deleteTarball();
+      throw err;
+    }
+
+    deleteTarball();
```

Comment:
```suggestion
    } catch (err) {
      // Avoid unhandled promise rejection
      throw err;
    } finally {
      deleteTarball();
    }
```

Reply:
I don't think this gonna work. `finally` will never be called, since you re-throw in the catch.

Reply:
Gonna merge and try. If does not work, will revert.

Reply:
![image](https://user-images.githubusercontent.com/6058569/102555174-3f85c180-4094-11eb-9c9e-3be00ac76431.png)